### PR TITLE
tile:remove method

### LINKED ISSUE #1522: Add tile:remove()
It's necessary to create huge random generated maps.

https://otland.net/threads/tfs-1-0-how-do-i-remove-a-tile.235513/#post-2273523

### PR DESCRIPTION
**Pull request by nekiro**
Resolve #1522

### CODE REVIEW

**Comment by Lyuzera** [comment]
nice, I think that with this method we can improve some things like loadMapChunk to replace all of the map and not duplicate.

**Comment by EvilHero90** [comment]
Aren't you forgetting to check for Creatures and remove them before removing the tile? (just quikcly looked over it)

**Comment by nekiro** [comment]
@EvilHero90 indeed, thanks!

**Review by EvilHero90 — DISMISSED** [review]
code wise this looks good now, havn't tested tho.

**Comment by ninjalulz** [comment]
You might want to set `flags` to `0` but other than that, it looks fine to me.

**Comment by nekiro** [comment]
@ninjalulz tile class will take care of that when removing creatures and then items.

**Comment by DSpeichert** [comment]
Could use a sample script to showcase how it's used (or useful).

File: src/map.cpp [hunk 1]
```diff
@@ -154,6 +154,44 @@ void Map::setTile(uint16_t x, uint16_t y, uint8_t z, Tile* newTile)
...
+	const Floor* floor = leaf->getFloor(z);
+	if (!floor) {
+		return;
+	}
+
+	Tile* tile = floor->tiles[x & FLOOR_MASK][y & FLOOR_MASK];
+	if (tile) {
+		if (const CreatureVector* creatures = tile->getCreatures()) {
+			for (int32_t i = creatures->size(); --i >= 0;) {
+				g_game.removeCreature((*creatures)[i]);
```
Comment (WibbenZ) [inline src/map.cpp:177]: Tested the code and works as it should, but not sure about this part.
What happens now is you can still see the ingame screen but your char is removed (just like when you use /r on a player)
Might be a good ide to kick the player and maybe even teleport it to temple, just tested to remove all the tiles around my char and got a client debug when I tried to log back in.

Comment (soul4soul) [reply src/map.cpp:177]: I like the idea of teleporting them to the temple in this odd case

**Review by WibbenZ — APPROVED** [review]
Removing merge block (not tested or read code)

**Comment by Znote** [comment]
Updated wiki
https://github.com/otland/forgottenserver/wiki/Metatable:Tile

**Pull request merged**